### DISCUSSION TITLE
How to specify the equivalent of `--no-default-features` to a `method = cargo` dependency ?

### DISCUSSION THREAD

**Discussion by yhql** [mesonbuild/meson#12377] [Q&A] [answered]
Hello, I am testing the recently merged `method = cargo` wrap method.

Declaring e.g.
```
bitflags_dep = dependency('bitflags-rs', version : '>=2.4.0')
```
with the appropriate wrap-file will generate a `meson.build` file that includes all dependencies although they are all optional.

Is there a way to specify that this dependency should be used with no specific feature currently? (`cargo`'s `--no-default-features` option)

**Comment by yhql** [comment]
Just realized that this is not exactly `--no-default-features`, rather the project `bitflags` itself has no dependency by default but meson will still produce a build file that requests all of the optional dependencies.

@xclaesse sorry to bother you directly but I think you are the one working on this :)

**Reply by yhql** [reply]
Seems this is fixed by https://github.com/mesonbuild/meson/pull/11867/

**Reply by eli-schwartz** [reply]
Yes, I believe so. You're just hitting a not-yet-implemented bit.

**Reply by xclaesse** [reply, marked as answer]
Indeed, current Cargo support is probably too limited to be useful unfortunately. It's a work in progress. Please give a try with https://github.com/mesonbuild/meson/pull/11867 or even https://github.com/mesonbuild/meson/pull/12363, hopefully they'll land for Meson 1.4.

**Reply by yhql** [reply]
Thanks for your feedback! Just for info I tried #11867 and can confirm it solves the problem.
I was trying to solve it in a basic way, that would require creating a `meson.options` file along with the `meson.build`, but I see your patch handles many more cases
